core/model/modx/modresource.class.php

```diff
@@ -450,6 +450,43 @@ function __construct(xPDO & $xpdo) {
         $this->_cacheKey= "[contextKey]/resources/[id]";
     }
 
+    /**
+     * Compute the "preview URL" for the resource
+     *
+     * @return string
+     */
+    public function getPreviewUrl() {
+        $url = '';
+        if (!$this->get('deleted')) {
+            $this->xpdo->setOption('cache_alias_map', false);
+            $sessionEnabled = '';
+            /** @var modContextSetting|null $ctxSetting */
+            $ctxSetting = $this->xpdo->getObject(
+                'modContextSetting',
+                array(
+                    'context_key' => $this->get('context_key'),
+                    'key' => 'session_enabled'
+                )
+            );
+
+            if ($ctxSetting) {
+                $sessionEnabled = $ctxSetting->get('value') == 0 ? array('preview' => 'true') : '';
```

Comment:
`$sessionEnabled` is already set to `''` a few lines earlier 😄.
```suggestion
                $sessionEnabled = array('preview' => 'true');
```

Reply:
From my understanding, if the setting is `1` (true) or anything else, we should not need to populate the query parameter.

Reply:
Something went wrong when I finished my review. I also shared another suggestion but that seems to be missing.